File: leetcode2025/2025_09/leetcode2025-09-22.py
from typing import List
from collections import Counter

class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        tot = list(Counter(nums).values())
        mx, cnt = 0, 0
        for v in tot:
            if v > mx:
                mx = v
                cnt = 1
            elif v == mx:
                cnt += 1
        return cnt*mx

        # cnt = sorted(list(Counter(nums).values()), reverse = True)
        # return cnt.count(cnt[0])*cnt[0]

if __name__ == "__main__":
    s = Solution()
    nums = [1,2,2,3,1,4]
    print(s.maxFrequencyElements(nums))

File: leetcode2025/2025_09/leetcode2025-09-22.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (int& i : nums){
            mp[i]++;
        }
        int mx = 0, cnt = 0;
        for (auto&kv : mp){
            if (kv.second > mx){
                mx = kv.second;
                cnt = 1;
            }
            else if (kv.second == mx){
                cnt++;
            }
        }
        return cnt*mx;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,2,2,3,1,4};
    cout << s.maxFrequencyElements(nums) << endl;
}
